src/test/regress/sql/functional_deps.sql

```diff
@@ -257,3 +260,143 @@ EXECUTE foo;
 ALTER TABLE articles DROP CONSTRAINT articles_pkey RESTRICT;
 
 EXECUTE foo;  -- fail
+
+-- Known issue (ADBDEV-4888) - in case there is a column in the target list
+-- that is not explicitly listed in the GROUP BY clause but has a functional
+-- dependency on another column in the GROUP BY clause, Postgres planner might
+-- output column values in grouped rows, that actually cannot be uniquely
+-- identified. As it can introduce instability in the tests, all the queries
+-- below are temporarily tested only with GPORCA.
+-- TODO: remove the hard-coded optimizer option below once ADBDEV-4888 is fixed.
+SET optimizer = on;
```

Comment:
> just not include the query in the tests 

I would like not to exclude any of the new test queries, as they cover the added functionality.
If any test query is removed there is high chance to break something if logic in FixGroupDependentTargets is changed.

> or show just a plan

If tests rely on plan, not on output, once somebody touches transfomation rules in GOPRCA, and another equivalent rule starts to apply, he might broke the tests (just because plan has changed, while the output did not).

> And mentioning internal ticket name in the comments is also unnecessary (unless we merge the ADBDEV-4888 asap and remove the comments).

It can be removed, we can leave a comment in the respective ticket in order not to forget to update the TODO during the ticket execution.